lib/pow/plug.ex

```diff
@@ -38,35 +39,54 @@ defmodule Pow.Plug do
 
   @doc """
   Put the provided config as a private key in the connection.
+
+  If `:namespace` is included in the configuration, then the configuration will
+  be stored in as private `:pow_config_NAMESPACE` key in the connection. This
+  key will be fetched in `fetch_config/1` if `:pow_namespace`
+  private key with the namespace value exists in the connection.
   """
   @spec put_config(Conn.t(), Config.t()) :: Conn.t()
   def put_config(conn, config) do
-    Conn.put_private(conn, @private_config_key, config)
+    Conn.put_private(conn, config_key(config), config)
+  end
+
+  defp config_key(%Conn{private: private}) do
+    config_key(@private_config_key, private[@private_namespace_key])
   end
+  defp config_key(config) do
+    config_key(@private_config_key, Config.get(config, :namespace))
+  end
+  defp config_key(key, nil), do: key
+  defp config_key(key, namespace), do: String.to_atom("#{key}_#{namespace}")
```

Comment:
I'm not sure about this one. I would prefer to not define atoms on runtime outside mix tasks.